src/Sarif.Driver.UnitTests/Sdk/EntryPointUtilitiesTests.cs

```diff
@@ -0,0 +1,154 @@
+        }
+
+        [Fact]
+        public void EnryPointUtilities_GenerateArguments_IgnoresNonexistentResponseFile()
+        {
+            const string ResponseFileName = "test.rsp";
+            var responseFileContents = new[] { "/b", "/c:val /d", "   /e   " };
+            var mockFileSystem = MakeMockFileSystem(ResponseFileName, responseFileContents);
+
+            const string NonexistentResponseFile = ResponseFileName + "x";
+            var args = new[] { "/a", "@" + NonexistentResponseFile, "/f" };
+
+            string[] result = EntryPointUtilities.GenerateArguments(args, mockFileSystem, MakeMockEnvironmentVariables());
+
+            result.Length.Should().Be(2);
+            result.Should().ContainInOrder("/a", "/f");
+        }
+
+        [Fact]
+        public void EnryPointUtilities_GenerateArguments_StripsQuotesFromAroundArgsWithSpacesInResponseFiles()
+        {
+            const string ResponseFileName = "test.rsp";
+            var responseFileContents = new[] { "a \"one two\" b" };
+            var mockFileSystem = MakeMockFileSystem(ResponseFileName, responseFileContents);
+
+            var args = new[] { "@" + ResponseFileName };
+
+            string[] result = EntryPointUtilities.GenerateArguments(args, mockFileSystem, MakeMockEnvironmentVariables());
+
+            result.Length.Should().Be(3);
+            result.Should().ContainInOrder("a", "one two", "b");
+        }
+
+        [Fact]
+        public void EnryPointUtilities_GenerateArguments_ExpandsEnvironmentVariablesInResponseFilePathName()
+        {
+            const string DirectoryVariableName = "InstallationDirectory";
+            const string DirectoryName = @"c:\MyDirectory";
+            var environmentVariableDictionary = new Dictionary<string, string>
+            {
+                { DirectoryVariableName, DirectoryName }
+            };
+            var mockEnvironmentVariables = MakeMockEnvironmentVariables(environmentVariableDictionary);
+
+            const string ResponseFileName = "test.rsp";
+
+            string responseFileNameArgument = String.Format(CultureInfo.InvariantCulture,
+                @"%{0}%\{1}", DirectoryVariableName, ResponseFileName);
+
+            string expandedResponseFileName = String.Format(CultureInfo.InvariantCulture,
+                @"{0}\{1}", DirectoryName, ResponseFileName);
+
+            var responseFileContents = new[] { "a", "b c" };
+            var mockFileSystem = MakeMockFileSystem(expandedResponseFileName, responseFileContents);
+
+            var args = new[] { "@" + responseFileNameArgument };
+
+            string[] result = EntryPointUtilities.GenerateArguments(args, mockFileSystem, mockEnvironmentVariables);
+
+            result.Length.Should().Be(3);
+            result.Should().ContainInOrder("a", "b", "c");
+        }
+
+        private static IFileSystem MakeMockFileSystem(string fileName, string[] fileContents)
```

Comment:
>MakeMockFileSystem [](start = 35, length = 18)

This and MakeMockEnvironmentVariables are generally useful. I'd suggest breaking them into a separate file with static class MakeMock and methods FileSystem and EnvironmentVariables (=> MakeMock.FileSystem). Or maybe MockFactory.MakeFileSystem.